src/stores/draw.store.ts

```diff
@@ -2,30 +2,67 @@ import { defineStore } from 'pinia'
 import { ref, Ref } from 'vue'
 
 import { getUid } from 'ol/util'
-import { DrawStateActive } from './draw.store.model'
+import { DrawStateActive, EditStateActive } from './draw.store.model'
 import { DrawnFeature } from '@/services/draw/drawn-feature'
 
 export const useDrawStore = defineStore('draw', () => {
   const activeFeatureId: Ref<String | undefined> = ref(undefined)
   const editingFeatureId: Ref<String | undefined> = ref(undefined)
   const drawStateActive = ref<DrawStateActive>(undefined)
-  // no immutable changes on drawnFeatures in functions bellow,
-  // but keep same Collection for sync with ol source (map)
+  const editStateActive = ref<EditStateActive>(undefined)
   const drawnFeatures = ref<DrawnFeature[]>([])
   const featureEditionDocked = ref(false)
 
-  function toggleActiveState(newState: DrawStateActive) {
-    if (drawStateActive.value === newState) {
+  function toggleDrawActiveState(newState: DrawStateActive) {
+    // allow draw of different geom type after edit, but not same type
+    if (editStateActive.value?.slice(4) === newState?.slice(4)) {
```

Comment:
I find this use of string type enums a bit risky.

What if a refactoring changed 'draw' into 'paint' and 'draw' and 'edit' do not have both a 4 Letter prefix.
At firest I thought, the 4 letter offset was shared between draw and edit

I would suggest 2 levels of fix:
- basic fix: replace slice by editStateActive.value?.replace('edit', '') and newState?.replace('draw', '') => this might be a bit more readable
- sophisticated fix: create a class with 2 arguments: feature type (point, label, circle, etc.) and feature context (draw, edit, select)